node-commons/src/main/java/org/apache/iotdb/commons/schema/node/role/IMeasurementMNode.java

```diff
@@ -16,23 +16,21 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-package org.apache.iotdb.db.metadata.mnode;
+package org.apache.iotdb.commons.schema.node.role;
 
 import org.apache.iotdb.commons.path.MeasurementPath;
+import org.apache.iotdb.commons.schema.node.IMNode;
 import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
 import org.apache.iotdb.tsfile.write.schema.IMeasurementSchema;
 
 /** This interface defines a MeasurementMNode's operation interfaces. */
-public interface IMeasurementMNode extends IMNode {
-
-  @Override
-  IEntityMNode getParent();
-
-  MeasurementPath getMeasurementPath();
+public interface IMeasurementMNode<N extends IMNode<N>> extends IMNode<N> {
 
   IMeasurementSchema getSchema();
 
-  TSDataType getDataType(String measurementId);
+  void setSchema(IMeasurementSchema schema);
```

Comment:
We assume the schema is immutable. Is this interface necessary?

Reply:
use to implement moveDataToNewMNode()

Reply:
There seems only device node replacement. There won't be any measurement node replacement. Maybe we can push down the moveDataToNewMNode() to only InternalMNode and its subclass.